Tests/AppStoreConnectClientTests/RetryingMiddlewareTests.swift

```diff
@@ -0,0 +1,250 @@
+        await fulfillment(of: [expectation], timeout: 3.0)
+    }
+    
+    func testNotRepeatWhenBodyIterationBehaviorSingle() async throws {
+        let sut = RetryingMiddleware(
+            signals: [.code(internalServerError)],
+            policy: .upToAttempts(count: 3)
+        )
+        let expectation = XCTestExpectation(
+            description: "Expect 'next' to be called only once due to 'single' iteration behavior"
+        )
+        expectation.expectedFulfillmentCount = 1
+        let next: @Sendable (
+            HTTPRequest,
+            HTTPBody?,
+            URL
+        ) async throws -> (HTTPResponse, HTTPBody?) = { _, _, _ in
+            expectation.fulfill()
+            let status = HTTPResponse.Status(code: self.internalServerError)
+            return (HTTPResponse(status: status), nil)
+        }
+        let baseURL = try XCTUnwrap(URL(string: "http://example.com"), "Unwrap fail")
+        let (response, _) = try await sut.intercept(
+            HTTPRequest(
+                method: .get,
+                scheme: "http",
+                authority: "example.com",
+                path: "/test"
+            ),
+            body: HTTPBody([1, 2, 3], length: .known(3), iterationBehavior: .single),
+            baseURL: baseURL,
+            operationID: "testOperation",
+            next: next
+        )
+        XCTAssertEqual(response.status.code, internalServerError)
+        await fulfillment(of: [expectation], timeout: 3.0)
+    }
+    
+    func testRepeatWhenIterationBehaviorMultipleAndPolicyRepeat() async throws {
+        let sut = RetryingMiddleware(
+            signals: [.code(internalServerError)],
+            policy: .upToAttempts(count: 3)
+        )
+        let expectation = XCTestExpectation(
+            description: "Expect 'next' to be called multiple times due to 'multiple' iteration behavior and retry policy"
+        )
+        expectation.expectedFulfillmentCount = 3
+        let next: @Sendable (
+            HTTPRequest,
+            HTTPBody?,
+            URL
+        ) async throws -> (HTTPResponse, HTTPBody?) = { _, _, _ in
+            expectation.fulfill()
+            let status = HTTPResponse.Status(code: self.internalServerError)
+            return (HTTPResponse(status: status), nil)
+        }
+        let baseURL = try XCTUnwrap(URL(string: "http://example.com"), "Unwrap fail")
+        let (response, _) = try await sut.intercept(
+            HTTPRequest(
+                method: .get,
+                scheme: "http",
+                authority: "example.com",
+                path: "/test"
+            ),
+            body: HTTPBody([1, 2, 3], length: .known(3), iterationBehavior: .multiple),
+            baseURL: baseURL,
+            operationID: "testOperation",
+            next: next
+        )
+        XCTAssertEqual(response.status.code, internalServerError)
+        await fulfillment(of: [expectation], timeout: 3.0)
```

Comment:
Set check after fulfillment

Reply:
I think you do not need timeout in this test

Reply:
if the test fails then the test hangs without a timeout.